### ISSUE TITLE
CMake install target seems broken

### ISSUE THREAD

**Issue by mikezackles** [ThePhD/sol2#585]
Hi! I'm trying to package sol2 for Arch Linux, but `make install` fails. It appears that the CMake build is trying to install the `include/sol2` directory, which doesn't exist. I also noticed that it doesn't install `sol.hpp`. The single-header and docs generation also look broken.

**Comment by ThePhD** [comment]
I'm new to this and any help making it better would be much appreciated.

**Comment by ThePhD** [comment]
Wait, why are Single and doc generation broken? All they require is Python 3. What's the output you're getting?

**Comment by mikezackles** [comment]
If I fix the `include/sol2` issue I get
```
CMake Error at cmake_install.cmake:72 (file):
  file INSTALL cannot find
  "/home/mikezackles/projects/aur/aur-sol2/src/sol2/build/./include/single/sol/sol.hpp".
```
and if I turn on docs I get
```
CMake Error at cmake_install.cmake:72 (file):
  file INSTALL cannot find
  "/home/mikezackles/projects/aur/aur-sol2/src/sol2/docs/build/html".
```
Also the docs command looks like it's trying to call `make html`?
I'm no CMake wizard myself, but happy to try to help sort it out, assuming I'm not just being dumb.

**Comment by mikezackles** [comment]
Build script is [here](https://github.com/mikezackles/aur-sol2/blob/master/PKGBUILD) for reference.

**Comment by ThePhD** [comment]
Yes, it's trying to call `make html` so it can build the HTML docs. It seems to be working okay for me, so I'm going to take another look at it on a few other platforms and make sure everything's okay...

**Comment by mikezackles** [comment]
I made some changes that fix the additional issues for me in the PR I submitted, though I have only verified it on my platform. Nothing triggers generating either the docs or the single header stuff on a make install, but it still tries to install the files and fails. Also for the single-header build, I think the `single.py` script is trying to place the generated files in a directory that doesn't exist without creating it first.
